testapp/tests/test_behaviour_for_postgres.py

```diff
@@ -102,3 +102,14 @@ def test_executing_multiple_querysets_which_are_always_empty_is_handled(self):
             self.assertEqual(len(results), 2)
             self.assertEqual(results[0], [])
             self.assertEqual(results[1], [])
+
+    def test_fetch_count(self):
+        with self.assertNumQueries(0):
+            store_product_count_queryset = StoreProduct.objects.filter()
+            store_product_count_queryset.fetch_count = True
```

Comment:
instead of modifying original queryset, lets make a wrapper util? something like 

```py
from service import QuerysetCountWrapper # gives way to lazily fetch count

querysets = [q1, q2, QuerysetCountWrapper(queryset=queryset)]
```

reason for this is, using `store_product_count_queryset` in some later stage might unintentionally fetch count?